frontend/www/js/omegaup/graderv2/CaseSelectorComponent.vue

```diff
@@ -0,0 +1,306 @@
+                }}<span class="score">{{
+                  score(groupResult(group.name))
+                }}</span></span
+              >
+              <span :title="group.name">{{ group.name }}</span>
+            </div>
+          </div>
+          <button
+            v-for="item in group.cases"
+            :key="item.name"
+            class="list-group-item list-group-item-action d-flex justify-content-between align-items-center"
+            type="button"
+            :class="{
+              'in-group': group.explicit,
+              active: currentCase == item.name,
+            }"
+            @click="selectCase(item.name)"
+          >
+            <div class="case-item">
+              <span
+                class="verdict"
+                :class="verdictClass(item.name)"
+                :title="verdictTooltip(caseResult(item.name))"
+                >{{ verdictLabel(caseResult(item.name))
+                }}<span class="score">{{
+                  score(caseResult(item.name))
+                }}</span></span
+              >
+              <span :title="item.name">{{ item.name }}</span>
+            </div>
+            <button
+              v-if="groups.length > 1"
+              aria-label="Close"
+              class="close"
+              type="button"
+              @click.prevent.stop="removeCase(item.name)"
+            >
+              <span aria-hidden="true">×</span>
+            </button>
+          </button>
+        </template>
+      </div>
+    </div>
+    <form @submit.prevent="createCase()">
+      <div class="input-group">
+        <input
+          v-model="newCaseWeight"
+          class="form-control case-weight"
+          type="text"
+        />
+        <input v-model="newCaseName" class="form-control" type="text" />
+        <div class="input-group-append">
+          <button
+            class="btn btn-secondary"
+            type="submit"
+            :disabled="newCaseName.length == 1"
+            @click="createCase()"
+          >
+            +
+          </button>
+        </div>
+      </div>
+    </form>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop } from 'vue-property-decorator';
+import * as Util from '../grader/util';
+
+@Component
+export default class MyComponent extends Vue {
+  @Prop({ type: Object, required: true }) store!: any;
+  @Prop({ type: Object, required: true }) storeMapping!: any;
+  @Prop({ type: String, default: 'vs-dark' }) theme!: string;
+
+  newCaseWeight: number = 1;
+  newCaseName: string = '';
+
+  get summary(): string {
+    if (
+      this.store.state.dirty ||
+      !this.store.state.results ||
+      !this.store.state.results.verdict
+    ) {
+      return '…';
+    }
+    return `${this.store.state.results.verdict} ${this.score(
+      this.store.state.results,
+    )}`;
+  }
+
+  get groups() {
+    const flatCases: any = Util.vuexGet(this.store, this.storeMapping.cases);
+    let resultMap: { [key: string]: any } = {};
```

Comment:
If `resultMap` will never change its value, can it be changed to a constant?

```suggestion
    const resultMap: { [key: string]: any } = {};
```

This can also be applied in almost all places where you use let